client/normal_array.go

```diff
@@ -41,6 +41,10 @@ func (v normalBoolArray) BoolArray() ([]bool, bool) {
 	return v.val, true
 }
 
+func (v normalBoolArray) IsEqual(other NormalValue) bool {
```

Comment:
nitpick: Whilst I prefer `IsEqual`, `Equal` seems to be the norm in Go (e.g. `bytes.Equal`), I suggest renaming these.

Reply:
done